daemons/fenced/fenced_commands.c

```diff
@@ -2674,16 +2681,15 @@ next_required_device(async_command_t *cmd)
 static void
 st_child_done(int pid, const pcmk__action_result_t *result, void *user_data)
 {
-    stonith_device_t *device = NULL;
     stonith_device_t *next_device = NULL;
     async_command_t *cmd = user_data;
+    stonith_device_t *device = cmd_device(cmd);
 
```

Comment:
as we have a check for cmd != NULL already we could leave setting device to where it was to
make explicit that without cmd we're not doing anything.

Reply:
might as well